generators/app/index.js

```diff
@@ -134,12 +159,34 @@ module.exports = class extends BaseGenerator {
         });
     }
 
+    getPreviousConfiguration() {
+        try {
+            return jsYaml.safeLoad(fsModule.readFileSync(`${jhipsterConstants.SERVER_MAIN_RES_DIR}config/application.yml`, 'utf8'));
+        } catch (e) {
+            this.log(
+                `${chalk.red.bold(
+                    'WARN!'
+                )} Could not parse the previous kafka configuration, some previous configurations could be overwritten\n`
+            );
+        }
+        return {};
+    }
+
     writing() {
         // function to use directly template
         this.template = function(source, destination) {
             this.fs.copyTpl(this.templatePath(source), this.destinationPath(destination), this);
         };
 
+        // function generate kafka application properties
+        this.generateKafkaProperties = function() {
+            let kafkaPropertiesWithEjs;
+            jhipsterUtils.renderContent(this.templatePath('src/main/resources/application-subPartKafka.yml.ejs'), this, this, {}, res => {
```

Comment:
We should keep only one `application.yml` and replace the `kafka:` block by the content of this file.

I would also replace the filename by `application-kafka.yml` even if it's temporary.

Reply:
this method use the ejs file to generate a content that i store in a variable to assume the replacing in the application yaml file